api/utils/uaaClient.js

```diff
@@ -0,0 +1,48 @@
+const request = require('request');
+const url = require('url');
+const config = require('../../config');
+
+class UAAClient {
+  constructor(accessToken) {
+    this.accessToken = accessToken;
+  }
+
+  async verifyUserGroup(userId, groupNames = []) {
+    const { groups, origin, verified } = await this.request('GET', `/Users/${userId}`);
+    if (origin === 'cloud.gov' && !verified) {
+      return false;
+    }
+
+    return groups.filter(group => groupNames.indexOf(group.display) > -1).length > 0;
```

Comment:
maybe the following would be more semantic:
```
return groups.some(group => groupNames.includes(group.display));
```

Reply:
Duly noted.